### ISSUE TITLE
Fix issue with memory helper in early 2.3

### ISSUE THREAD

**Issue by fooman** [extdn/github-actions-m2#18]
There are a couple of M2 core issues

2.3.0 is broken due to
```
In Filesystem.php line 156:

  [Magento\Framework\Exception\LocalizedException]
  The XML in file "/tmp/m2/vendor/magento/module-inventory-catalog/etc/commun
  ication.xml" is invalid:
  Element 'topic', attribute 'is_synchronous': The attribute 'is_synchronous'
   is not allowed.
  Line: 9

  Element 'topic', attribute 'is_synchronous': The attribute 'is_synchronous'
   is not allowed.
  Line: 12

  Element 'topic', attribute 'is_synchronous': The attribute 'is_synchronous'
   is not allowed.
  Line: 15

  Element 'topic', attribute 'is_synchronous': The attribute 'is_synchronous'
   is not allowed.
  Line: 18

  Element 'topic', attribute 'is_synchronous': The attribute 'is_synchronous'
   is not allowed.
  Line: 21

  Verify the XML and try again.
```
see https://github.com/extdn/extension-dashboard-m2/runs/1312019553?check_suite_focus=true

2.3.1 - 2.3.3
broken due to 
```
OK (4 tests, 5 assertions)

Fatal error: Uncaught PHPUnit\Framework\Exception: Notice: Undefined variable: e in /tmp/m2/dev/tests/integration/framework/Magento/TestFramework/Helper/Memory.php:52.

/tmp/m2/dev/tests/integration/framework/bootstrap.php:134
/tmp/m2/dev/tests/integration/framework/Magento/TestFramework/Helper/Memory.php:52
/tmp/m2/dev/tests/integration/framework/Magento/TestFramework/MemoryLimit.php:127
/tmp/m2/dev/tests/integration/framework/Magento/TestFramework/MemoryLimit.php:59
/tmp/m2/dev/tests/integration/framework/Magento/TestFramework/Bootstrap/Memory.php:50

  thrown in /tmp/m2/dev/tests/integration/framework/bootstrap.php on line 134
```
https://github.com/extdn/extension-dashboard-m2/runs/1312039797?check_suite_focus=true

see https://github.com/extdn/extension-dashboard-m2/runs/1312423133?check_suite_focus=true

**Comment by fooman** [comment]
@jissereitsma I believe you already ran into this prior, but it looks like the patch we apply may be for something else?

Thinking that maybe we exclude the memory test from the shipped phpunit.xml?

**Comment by jissereitsma** [comment]
The broken 2.3.0 run seems due to a wrong MSI package being published somewhere. Not sure how to fix this, except from removing that `module-inventory-catalog` XML file manually from the installation of 2.3.0.

The memory issue with 2.3.1 to 2.3.3 is indeed an issue that I fixed earlier with a patch here: https://github.com/extdn/github-actions-m2/blob/master/magento-integration-tests/entrypoint.sh#L110 But it was not due to a single test, it was due to the reporting of memory usage of all tests, after running all tests. The code of this memory helper contains some dark magic for Window vs Linux, but still made wrong assumptions. However, I believe that error explicitly mentioned `tasklist.exe` (a Windows binary) not to be found on the Linux-based CI containers. And this error seems to be dealing with something else. However, the same methods are involved, so my earlier patch might work right away.
